File: 0724/solution.ts
function pivotIndex(nums: number[]): number {
  let sum = 0;
  for (let num of nums) {
    sum += num;
  }
  let left = sum;
  let right = 0;
  for (let i = 0; i < nums.length; i++) {
    left -= nums[i];
    if (left === right) {
      return i;
    }
    right += nums[i];
  }
  return -1;
}

File: 0724/solution.java
class Solution {
    public int pivotIndex(int[] nums) {
        if (nums.length == 1) return 0;
        int right = 0;
        for(int num: nums) {
            right += num;
        }
        int left = 0;
        for(int i = 0; i < nums.length; i++) {
            right -= nums[i];
            if (left == right) return i;
            left += nums[i];
        }
        return -1;
    }
}